### PR TITLE
Add Firebase authentication.

### PR DESCRIPTION
**Pull request by leerob**

### CODE REVIEW

**Comment by leerob** [comment]
#### 📝Changed routes:


| [**/**](https://fastfeedback-git-firebase-auth.leerob1.now.sh/) |
|:-:|
|<a href="https://fastfeedback-git-firebase-auth.leerob1.now.sh/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://s.microlink.io/2fNnO9uP2M0erS9t2FEr2OWlGGQ_.png">(view full size)</a> |







Commit 40a6858574ad73fa1d261e6baab57b36abfbec9d (https://fastfeedback-git-firebase-auth.leerob1.now.sh).

**Comment by leerob** [comment]
For those following along: I rotated all the secrets and added the new environment variables in Vercel's dashboard.

File: lib/auth.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        setUser(false);
+      });
+  };
+
+  useEffect(() => {
+    const unsubscribe = firebase.auth().onAuthStateChanged((user) => {
+      if (user) {
+        setUser(user);
+      } else {
+        setUser(false);
```
Comment (fenske) [inline lib/auth.js:42]: @leerob was curious why `setUser(false)` and not `setUser(null)`?

Comment (leerob) [reply lib/auth.js:42]: Good question! Either could work 😄

**Pull request merged**